x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/logical/And.java

```diff
@@ -36,4 +36,17 @@ public And swapLeftAndRight() {
     public Or negate() {
         return new Or(location(), new Not(location(), left()), new Not(location(), right()));
     }
+
+    @Override
+    public boolean nullable() {
+        if (left().nullable() && right().foldable() && right().fold() == Boolean.TRUE) {
```

Comment:
Maybe add a static package method on `BinaryLogic` to avoid doing `left/right.foldable() && fold = ?` and do `if (maybeFoldsTo(right(), Boolean.TRUE)` etc...

Reply:
yep, forgot that, extracting method to `Foldables`